framework/include/loops/NonlinearThread.h

```diff
@@ -152,9 +153,22 @@ class NonlinearThread : public ThreadedElementLoop<ConstElemRange>
   MooseObjectWarehouse<KernelBase> * _tag_kernels;
   ///@}
 
+  ///@{
+  /// Reference to HDGKernel storage structures
+  MooseObjectTagWarehouse<HDGKernel> & _hdg_kernels;
+
+  MooseObjectWarehouse<HDGKernel> * _hdg_warehouse;
+  ///@}
+
   /// Current subdomain FVElementalKernels
   std::vector<FVElementalKernel *> _fv_kernels;
 
   /// Whether there are any active residual objects; otherwise we will do an early return
   const bool _has_active_objects;
+
+private:
+  /// Whether DG kernels should be executed for a given internal side. This can be false if HDG
+  /// kernels are present because in that case we visit the internal side from both sides and we
+  /// don't want to double execute DG kernels
```

Comment:
the logic seems to not match this comment?
>   return _should_execute_dg || _hdg_warehouse->hasActiveBlockObjects(_subdomain, _tid);

or do you mean "should execute dg" is false but that's ok for `shouldComputeInternalSide` because we also check the warehouse and will execute anyway?

Reply:
I've tried to clarify the comment